tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,527 @@
+    a duplicate dataset with a new name. If the template dataset is not provided, the user must
+    define the primary_keys for the dataset to be created. Additional attributes can be added
+    in the attributes argument. The default attribute type will be ARRAY STRING. To set non-default
+    attribute types, you must define them in the attribute_types dictionary. Any attribute
+    descriptions can be specified in the attribute_descriptions dictionary.
+
+    Args:
+        client: TUC client object
+        dataset_name: name for the new dataset being created
+        dataset: optional dataset TUC object to use as a template for the new dataset
+        primary_keys: one or more attributes for primary key(s) of the new dataset
+        attributes: a list of attribute names to create in the new dataset
+        attribute_types: dictionary for non-default types, attribute name is the key and
+            AttributeType is the value
+        attribute_descriptions: dictionary for attribute descriptions, attribute name is the key
+            and the attribute description is the value
+        description: description of the new dataset
+        external_id: external_id for dataset, if None Tamr will create one for you
+        tags: the list of tags for the new dataset
+
+    Returns:
+        Dataset created in Tamr
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        ValueError: If both dataset and primary_keys are not defined
+        ValueError: If the dataset already exists
+        TypeError: If the attributes argument is not an Iterable
+
+    Example:
+        >>> import tamr_toolbox as tbox
+        >>> tamr_client = tbox.utils.client.create(**instance_connection_info)
+        >>> tbox.dataset.manage.create(
+        >>>     client=tamr_client,
+        >>>     dataset_name="my_new_dataset",
+        >>>     primary_keys=["unique_id"],
+        >>>     attributes=["name","address"],
+        >>>     description="My new dataset",
+        >>> )
+    """
+
+    if not dataset and not primary_keys:
+        raise ValueError(f"dataset or primary_keys must be defined")
+
+    # Get dataset information
+    if dataset:
+        # Get attributes from dataset object
+        attribute_types, attribute_descriptions = {}, {}
+        for attr in dataset.attributes.stream():
+            attribute_types[attr.name] = attribute_type.from_json(attr.type.spec().to_dict())
+            attribute_descriptions[attr.name] = attr.description
+        attributes = attribute_types.keys()
+        # Get dataset spec information
+        description = dataset.description
+        tags = dataset.tags
+        primary_keys = dataset.key_attribute_names
+
+    # Check input type is correct
+    if attributes and not isinstance(attributes, Iterable):
+        raise TypeError("attributes arg must be an Iterable")
+
+    dataset_exists = exists(client=client, dataset_name=dataset_name)
+    if not dataset_exists:
```

Comment:
nit: can be more concise since this variable isn't reused
```suggestion
    if not exists(client=client, dataset_name=dataset_name):
```